ui/jetbrains/src/main/java/com/neueda/jetbrains/plugin/graphdb/jetbrains/ui/datasource/tree/TreeNodeModel.java

```diff
@@ -0,0 +1,95 @@
+package com.neueda.jetbrains.plugin.graphdb.jetbrains.ui.datasource.tree;
+
+import com.neueda.jetbrains.plugin.graphdb.jetbrains.component.datasource.state.DataSourceApi;
+import com.neueda.jetbrains.plugin.graphdb.jetbrains.ui.datasource.metadata.dto.ContextMenu;
+
+import javax.swing.*;
+import java.util.Optional;
+
+public class TreeNodeModel implements TreeNodeModelApi {
+
+    private ContextMenu contextMenu;
+    private NodeType type;
+    private Icon icon;
+    private String value;
+    private DataSourceApi dataSourceApi;
+
+    public TreeNodeModel(Neo4jTreeNodeType type, String value) {
+        this.type = type;
+        this.value = value;
+    }
+
+    public TreeNodeModel(Neo4jTreeNodeType type, DataSourceApi dataSourceApi, String value) {
```

Comment:
Constructors should be organized into a tree, with `prepareContextMenu()` only in one of them (most general one).